File: client-ios/Sources/Core/Models/RealtimeCallStats.swift
import Foundation

struct RealtimeCallStats: Equatable {
    var transportPath: String?
    var rttMs: Double?
    var availableOutgoingKbps: Double?
    var audioRxPacketLossPct: Double?
    var audioTxPacketLossPct: Double?
    var audioJitterMs: Double?
    var audioPlayoutDelayMs: Double?
    var audioConcealedPct: Double?
    var audioRxKbps: Double?
    var audioTxKbps: Double?
    var videoRxPacketLossPct: Double?
    var videoTxPacketLossPct: Double?
    var videoRxKbps: Double?
    var videoTxKbps: Double?
    var videoFps: Double?
    var videoResolution: String?
    var videoFreezeCount60s: Int64?
    var videoFreezeDuration60s: Double?
    var videoRetransmitPct: Double?
    var videoNackPerMin: Double?
    var videoPliPerMin: Double?
    var videoFirPerMin: Double?
    var updatedAtMs: Int64 = 0

    static let empty = RealtimeCallStats()
}

File: client-android/app/src/main/java/app/serenada/android/call/RealtimeCallStats.kt
package app.serenada.android.call

data class RealtimeCallStats(
    val transportPath: String? = null,
    val rttMs: Double? = null,
    val availableOutgoingKbps: Double? = null,
    val audioRxPacketLossPct: Double? = null,
    val audioTxPacketLossPct: Double? = null,
    val audioJitterMs: Double? = null,
    val audioPlayoutDelayMs: Double? = null,
    val audioConcealedPct: Double? = null,
    val audioRxKbps: Double? = null,
    val audioTxKbps: Double? = null,
    val videoRxPacketLossPct: Double? = null,
    val videoTxPacketLossPct: Double? = null,
    val videoRxKbps: Double? = null,
    val videoTxKbps: Double? = null,
    val videoFps: Double? = null,
    val videoResolution: String? = null,
    val videoFreezeCount60s: Long? = null,
    val videoFreezeDuration60s: Double? = null,
    val videoRetransmitPct: Double? = null,
    val videoNackPerMin: Double? = null,
    val videoPliPerMin: Double? = null,
    val videoFirPerMin: Double? = null,
    val updatedAtMs: Long = 0L
)
